### PR TITLE
Add idomatic rust trts crate

### PR DESCRIPTION
**Pull request by nick-mobilecoin**

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #125** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/125" target="_blank">Graphite</a> 
    * **PR #126** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/126" target="_blank">Graphite</a> 
      * **PR #127** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/127" target="_blank">Graphite</a> 
        * **PR #130** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/130" target="_blank">Graphite</a> 
          * **PR #131** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/131" target="_blank">Graphite</a> 
            * **PR #133** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/133" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/133?utm_source=stack-comment).

File: trts/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    /// - `item` - The item to see if it's fully outside of the enclave's memory
+    ///   space
+    fn is_outside_enclave(item: &T) -> bool;
+}
+
+impl<T: Sized> EnclaveMemory<T> for T {
+    fn is_within_enclave(item: &T) -> bool {
+        let start = item as *const _ as *const c_void;
+        let size = mem::size_of::<T>();
+        matches!(unsafe { sgx_is_within_enclave(start, size) }, 1)
```
Comment (awygle) [inline trts/src/lib.rs:31]: Why use `matches!` here?
```suggestion
        unsafe { sgx_is_within_enclave(start, size) } == 1
```

File: trts/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    }
+
+    fn is_outside_enclave(item: &T) -> bool {
+        let start = item as *const _ as *const c_void;
+        let size = mem::size_of::<T>();
+        matches!(unsafe { sgx_is_outside_enclave(start, size) }, 1)
```
Comment (awygle) [inline trts/src/lib.rs:37]: Why use `matches!` here?
```suggestion
        unsafe { sgx_is_outside_enclave(start, size) } == 1
```

**Review by awygle — APPROVED** [review]
LGTM, I just think `matches!` is a bit odd (I had to google what it did).

File: trts/src/lib.rs (hunk 1 above)
Comment (nick-mobilecoin) [reply trts/src/lib.rs:31]: yah, I wasn't too fond of `matches!` either.
I started with
```rust
        match unsafe { sgx_is_within_enclave(start, size) } {
            1 => true,
            _ => false,
        }
```
Then clippy gave me this
```shell
   |
31 | /         match unsafe { sgx_is_within_enclave(start, size) } {
32 | |             1 => true,
33 | |             _ => false,
34 | |         }
   | |_________^ help: try this: `matches!(unsafe { sgx_is_within_enclave(start, size) }, 1)`
   |
   = note: `#[warn(clippy::match_like_matches_macro)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#match_like_matches_macro
```
I've updated them to 
```rust
        let result = unsafe { sgx_is_within_enclave(start, size) };
        result == 1
```
one can't do 
```rust
unsafe { sgx_is_within_enclave(start, size) } == 1
```
but one can do
```rust
1 == unsafe { sgx_is_within_enclave(start, size) }
```

I felt the 2 lines was more intuitive and with rust's non mut by default one doesn't need to worry, as much, about C's `if(x = CONSTANT)` error.

**Review by jcape — APPROVED** [review]

**Pull request merged**